### ISSUE TITLE
Don't use clusters for quick tests

### ISSUE THREAD

**Issue by gjoseph92** [coiled/coiled-runtime#420]
Some tests have been added recently (and some have existed for a while) that run extremely quickly (<10s). Though they may be valuable tests, I don't think they should run against a coiled cluster. There's too much variance to get a useful signal from them.

These tests should either:
1. just use a local cluster
2. repeat many times

https://github.com/coiled/coiled-runtime/blob/f81cf53151c7fb8e1bbcb12e781053f7ca94e9ef/tests/benchmarks/test_futures.py#L6-L8

https://github.com/coiled/coiled-runtime/blob/f81cf53151c7fb8e1bbcb12e781053f7ca94e9ef/tests/benchmarks/test_array.py#L161-L177

https://github.com/coiled/coiled-runtime/blob/f81cf53151c7fb8e1bbcb12e781053f7ca94e9ef/tests/benchmarks/test_zarr.py#L14-L17

**Comment by ntabris** [comment]
> Though they may be valuable tests, I don't think they should run against a coiled cluster. There's too much variance to get a useful signal from them.

Variance because of Coiled? My understanding is that test time doesn't include time to spin up cluster, so this would surprise me a little.

Could we move the super fast tests into the same file so (I think) they'd all run on the same cluster?

**Comment by jrbourbeau** [comment]
Yeah, I'd also like to have a better understanding of what variance @gjoseph92 is talking about

**Comment by gjoseph92** [comment]
On a recent AB test with 4 repeats, the standard deviation within a test case was larger than the differences between test cases:

![Screen Shot 2022-10-04 at 2 19 11 PM](https://user-images.githubusercontent.com/3309802/193918556-8eb9d0f9-90e8-4b49-9517-0aba8a7a852f.png)
![Screen Shot 2022-10-04 at 2 19 37 PM](https://user-images.githubusercontent.com/3309802/193918553-c7d18c38-03c4-4c77-ab97-be558b4b353b.png)
![Screen Shot 2022-10-04 at 2 20 04 PM](https://user-images.githubusercontent.com/3309802/193918558-ddc56ea8-f43a-48d6-817d-e5518db41307.png)

Compare that to a long-running test like `test_shuffle`, where the effect size is much larger than the noise within a test case:
![image](https://user-images.githubusercontent.com/3309802/193918990-24a1631f-2fbd-4bcd-b040-6dc1578bdcc0.png)

**Comment by gjoseph92** [comment]
> Variance because of Coiled? My understanding is that test time doesn't include time to spin up cluster, so this would surprise me a little.

Probably not "because of Coiled", as in not Coiled's fault. There's just way more complexity and natural variance involved in connecting to another machine over the internet which then connects to more machines. (There's still plenty of variance in a local cluster too; maybe even too much. I just bet there's less.)

I'm not too concerned about the variance and don't think we should put much effort into investigating it right now.

My point is just that a single iteration of a really fast test is not very accurate. For the same reason that `%timeit` runs your microbenchmarks 1000s of times and averages them, we should also run these sorts of micro-benchmarks many times. The size of the effect we are trying to measure is going to be small compared to the power of our measurement.

Hence why I say we shouldn't use clusters for fasts tests. Instead, we should either:
1. not use a cluster (to remove a source of variance)
2. make the tests not fast (loop them enough times that the effect size would become visible)

**Comment by fjetter** [comment]
I think the `test_single_future` is maybe the best example. This would be great in an asv suite like https://github.com/dask/dask-benchmarks

I think another interesting example is https://github.com/coiled/coiled-runtime/issues/339 which is measuring performance to read a single parquet file. There is not a real reason why we'd need to measure this on a coiled cluster.

My concern in not dispatching this to a coiled cluster is mostly that I don't trust GH actions runners to have reliable performance. Ideally, we'd have a simple VM to run these tests in isolation
